### PR TITLE
DR-1361 recover same pod

### PR DESCRIPTION
**Pull request by ddietterich**
At the current pod name to the list of obsolete pods on initialization. The idea is that if kubernetes restarts the container because it thinks it is misbehaving, this will convince stairway to recover the flights that were abandoned due to that restart.

I worked hard to generate a test that would emulate the kind of restart we have been seeing with Raaid's soft delete workload. Despite my best attempts, I cannot get exactly the same failure. I do believe i have demonstrated that it does restart properly, at the failure symptoms are different.

### CODE REVIEW

File: datarepo-clienttests/src/main/java/scripts/testscripts/BulkLoad.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
   /** Public constructor so that this class can be instantiated via reflection. */
   public BulkLoad() {
     super();
-    manipulatesKubernetes = true; // this test script manipulates Kubernetes
+    // manipulatesKubernetes = true; // this test script manipulates Kubernetes
```
Comment (nmalfroy) [inline datarepo-clienttests/src/main/java/scripts/testscripts/BulkLoad.java]: Commented code

**Comment by fboulnois** [comment]
> Is this Florian's changes from earlier?

Looks like it is but doesn't appear in the final changeset (in terms of Dan's changes). Maybe from a rebase due to changes in `UnauthenticatedApiController.java`?

File: src/main/java/bio/terra/service/job/JobService.java [hunk 2]
```diff
@@ -147,6 +147,11 @@ public void initialize() {
                 }
             }
 
+            // Add our own pod name to the list of obsolete stairways. Sometimes Kubernetes will
```
Comment (fboulnois) [inline src/main/java/bio/terra/service/job/JobService.java:150]: This comment is really useful in addition to your description, thanks!

**Review by fboulnois — APPROVED** [review]
lgtm! 🙏

File: datarepo-clienttests/src/main/java/scripts/testscripts/BulkLoad.java (hunk 1 above)
Comment (ddietterich) [reply datarepo-clienttests/src/main/java/scripts/testscripts/BulkLoad.java]: Thanks for catching that. I still cannot run the tool in my environment without disabling manipulateKubernetes...

**Review by nmalfroy — APPROVED** [review]
lgtm

**Pull request merged**